backend/Controller.js

```diff
@@ -1,128 +1,173 @@
-                    User.findOneAndDelete({ "userID": data.body.id }, { "sort": { "userID": 1 } })
-                        .then(() => {
-                            console.log("ID of login attempt: " + data.body.id);
-
-                            //create new User in MongoDB
-                            User.create(newUser).then(
-                                data => {
-                                    res.json(data);
-                                },
-                                err => {
-                                    res.json(err);
-                                }
-                            );
-                        },
-                            //error in user delete
-                            err => {
-                                res.json(err);
-                            }
-                        );
+          //update user object with new information, and create if not already existing
+          User.updateOne(
+            { userID: data.body.id }, //Filter
+            {
+              $set: {
+                userID: data.body.id,
+                refreshToken: req.body.refreshToken,
+                name: data.body.display_name,
+                imageURL: data.body.images[0].url,
+                email: data.body.email,
+              },
+            }, //Update
+            { upsert: true } //create User if does not already exist
+          )
+            .then(() => {
+              res.json({
+                message: "User logged in successfully.",
+                return: {
+                  userID: data.body.id,
+                  refreshToken: req.body.refreshToken,
+                  name: data.body.display_name,
+                  imageURL: data.body.images[0].url,
+                  email: data.body.email,
                 },
-                //SpotifyAPI return error
-                (err) => {
-                    res.status(400).json(err);
-                })
+              });
+            })
+            .catch((err) => {
+              res.json({ message: "Unable to login user.", error: err });
+            });
         },
-        //access token refresh error
+        //SpotifyAPI return error
         (err) => {
-            res.json("Unable to refresh access token.");
-            console.log(err);
+          res.json({ message: "SpotifyAPI return error.", error: err });
         }
-    );
-}
+      );
+    },
+    //access token refresh error
+    (err) => {
+      res.json({ message: "Unable to refresh access token.", error: err });
+    }
+  );
+};
+
+//middleware for group joining endpoint
+exports.joinGroup = async (req, res) => {
+  //add userID to the group object
+  Group.updateOne(
+    { groupCode: req.body.groupCode }, //filter
+    { $addToSet: { users: req.body.spotifyID } }
+  )
+    .then(() => {
+      //add groupCode to user object
+      User.updateOne(
+        { userID: req.body.spotifyID }, //filter
+        { $addToSet: { groups: req.body.groupCode } }
```

Comment:
NOTE: I can join a group I already exist in. Should probably introduce a check that disables users from joining a group they are already a part of